tests/bin/demo-addition.test.ts

```diff
@@ -0,0 +1,55 @@
+import main from '@/bin/demo-addition';
+
+const maybeMockRestore = (a: any): void =>
+  a.mockRestore && typeof a.mockRestore === 'function'
+    ? a.mockRestore()
+    : undefined;
+
+type FunctionPropertyNames<T> = {
+  [K in keyof T]: T[K] extends (...args: any[]) => any ? K : never;
+}[keyof T];
+type ArgsType<T> = T extends (...args: infer A) => any ? A : never;
+
+function spyOnImplementing<
+  T extends typeof console,
+  M extends FunctionPropertyNames<T>,
+  F extends T[M],
+  I extends (...args: any[]) => ReturnType<F>
+>(
+  target: T,
+  property: M,
+  impl: I,
+): jest.SpyInstance<ReturnType<F>, ArgsType<F>> {
+  maybeMockRestore(target[property]);
+  return jest.spyOn(target, property as any).mockImplementation(impl);
+}
+
+const mockConsoleLog = () => spyOnImplementing(console, 'log', (() => ({})));
```

Comment:
I realised after pushing the tests to this branch that the code wasn't style compliant. I've since fixed almost everything with the latest commit, however I'm still having issues with this line, in particular `npm run lint` wants to change `() => ({})` to `) => ({}`, which breaks the code. I might try defining a function that's just an empty arrow function and putting that in there to see if that fixes the style issues.

Reply:
This doesn't quite look right. Mocking the console.log is a simple thing, there's no need to create so many utility functions.

Reply:
Furthermore, when you test the main process, it's not even using console.log.

It's using STDOUT.

Therefore if you are using STDOUT, you must use the jest-mock-process, not the console.log mocking.

The process.stdout and console.log are 2 totally different things.